src/objects/screen.c

```diff
@@ -781,73 +806,153 @@ static int luaC_screen_set_position(lua_State *L)
     return 0;
 }
 
+/** Set the screen mode.
+ *
+ * @method get_modes
+ * @treturn int[][] modes Array or arrays containing width, height and refresh rate
+ */
+static int luaC_screen_get_modes(lua_State *L)
+{
+    struct cwc_output *output = luaC_screen_checkudata(L, 1);
+
+    struct wlr_output_mode *mode;
+    lua_newtable(L);
+    int i = 1;
+    wl_list_for_each(mode, &output->wlr_output->modes, link)
+    {
+        lua_newtable(L);
```

Comment:
The size of the table is known better to use lua_createtable.
```suggestion
        lua_createtable(L, 3, 0);
```